client/src/components/DetailView.tsx

```diff
@@ -172,44 +195,56 @@ class DetailViewWithoutI18n extends Component<Props, State> {
                           </ul>
                         </div>
                       </div>
-                      <div className="card-body-landlord">
-                        <div className="columns">
-                          <div className="column col-xs-12 col-6">
+                      {detailAddr.allcontacts && (
+                        <div className="card-body-landlord">
+                          <div className="card-title-landlord">
                             <b>
-                              <Trans>Business Entities</Trans>
+                              <Trans>Who’s the landlord of this building?</Trans>
                             </b>
-                            <ul>
-                              {detailAddr.corpnames &&
-                                detailAddr.corpnames.map((corp, idx) => <li key={idx}>{corp}</li>)}
-                            </ul>
+                            <LearnMoreAccordion />
                           </div>
-                          <div className="column col-xs-12 col-6">
-                            <b>
-                              <Trans>Business Addresses</Trans>
-                            </b>
-                            <ul>
-                              {detailAddr.businessaddrs &&
-                                detailAddr.businessaddrs.map((rba, idx) => (
-                                  <li key={idx}>{rba}</li>
-                                ))}
-                            </ul>
-                          </div>
-                        </div>
-                        {ownernames && (
                           <div>
-                            <b>
-                              <Trans>People</Trans>
-                            </b>
-                            <ul>
-                              {ownernames.map((owner, idx) => (
-                                <li key={idx}>
-                                  {owner.title.split(/(?=[A-Z])/).join(" ")}: {owner.value}
-                                </li>
-                              ))}
-                            </ul>
+                            {
+                              // Group all contact info by the name of each person/corporate entity
+                              Object.entries(_groupBy(detailAddr.allcontacts, "value"))
+                                // Move head officers and owners to the top of the list
+                                .sort((contact) =>
+                                  contact[1].find(
+                                    (o) => o.title === "HeadOfficer" || o.title.includes("Owner")
+                                  )
+                                    ? -1
+                                    : 0
+                                )
```

Comment:
It might make the code more readable to factor this out into a separate well-named function, e.g. `sortContactsByImportance()` or something?  That would also make it easier to unit test and such, if someone ever wanted to do that.